src/main/java/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package service.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import service.OperationHandler;
+import service.ShopService;
+import strategy.OperationStrategy;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Map<String, Integer> storage = new HashMap<>();
```

Comment:
let's move your storage to the separate class so that decouple shopservice and your storage